service/controller/resource/s3object/current.go

```diff
@@ -35,13 +35,16 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// anything here anymore. The current implementation relies on the bucket
 	// deletion of the s3bucket resource, which deletes all S3 objects and the
 	// bucket itself.
-	if key.IsDeleted(cr) {
-		if cc.Status.TenantCluster.Encryption.Key == "" {
+	if cc.Status.TenantCluster.Encryption.Key == "" {
```

Comment:
Due to the separation of the encryption resources it is now more important to ensure the encryption key is actually transported with the controller context.